### PR TITLE
WIP: preliminary RHEL for Edge support

### PR DESCRIPTION
**Pull request by teg**
This is completely untested, just putting it up here on popular demand.

### CODE REVIEW

File: internal/distro/rhel8/distro.go [hunk 1]
```diff
@@ -531,10 +532,85 @@ func qemuAssembler(format string, filename string, uefi bool, imageOptions distr
...
 // New creates a new distro object, defining the supported architectures and image types
 func New() distro.Distro {
 	const GigaByte = 1024 * 1024 * 1024
 
+	edgeImgType := imageType{
+		name:     "rhel-edge-commit",
+		filename: "commit.tar",
+		mimeType: "application/x-tar",
+		packages: []string{
+			"rhel-release", // TODO: is this correct for Edge?
```
Comment (mrguitar) [inline internal/distro/rhel8/distro.go:560]: change to redhat-release

File: internal/distro/rhel8/distro.go [hunk 2]
```diff
@@ -531,10 +532,85 @@ func qemuAssembler(format string, filename string, uefi bool, imageOptions distr
...
+			"glibc", "glibc-minimal-langpack", "nss-altfiles",
+			"sssd-client", "libsss_sudo", "shadow-utils",
+			"kernel",
+			"dracut-config-generic", "dracut-network",
+			"rpm-ostree", "polkit", "lvm2",
+			"chrony", "zram",
```
Comment (mrguitar) [inline internal/distro/rhel8/distro.go:566]: I don't believe zram is available in rhel. Please remove or comment out for now.

File: internal/distro/rhel8/distro.go [hunk 3]
```diff
@@ -531,10 +532,85 @@ func qemuAssembler(format string, filename string, uefi bool, imageOptions distr
...
+			"cryptsetup", "pinentry",
+			"keyutils",
+			"e2fsprogs", "dosfstools",
+			"gnupg2",
+			"basesystem", "python3", "bash",
```
Comment (mrguitar) [inline internal/distro/rhel8/distro.go:571]: python3 is platform-python in rhel8 (unless you want to 36 default module). I think we can leave it out and the right thing will get pulled in by deps

File: internal/distro/rhel8/distro.go [hunk 4]
```diff
@@ -531,10 +532,85 @@ func qemuAssembler(format string, filename string, uefi bool, imageOptions distr
...
+			"xz", "gzip",
+			"coreutils", "which", "curl",
+			"firewalld", "iptables",
+			"NetworkManager", "NetworkManager-wifi", "NetworkManager-wwan",
+			"wpa_supplicant", "iwd",
```
Comment (mrguitar) [inline internal/distro/rhel8/distro.go:576]: iwd doesn't exist in 8.2. Remove or comment out.

File: internal/distro/rhel8/distro.go [hunk 5]
```diff
@@ -531,10 +532,85 @@ func qemuAssembler(format string, filename string, uefi bool, imageOptions distr
...
+			"dnsmasq", "traceroute",
+			"hostname", "iproute", "iputils",
+			"openssh-clients", "openssh-server", "passwd",
+			"policycoreutils", "procps-ng", "rootfiles", "rpm",
+			"selinux-policy-targeted", "setup", "shadow-utils",
+			"sudo", "systemd", "util-linux", "vim-minimal",
+			"less", "tar",
+			"fwupd", // "usbguard", BUG: this fails due to an SELinux policy issue
+			"greenboot", "greenboot-grub2", "greenboot-rpm-ostree-grub2", "greenboot-reboot", "greenboot-status",
```
Comment (mrguitar) [inline internal/distro/rhel8/distro.go:585]: we definitely want greenboot, but please comment out for testing.

File: internal/distro/rhel8/distro.go [hunk 6]
```diff
@@ -531,10 +532,85 @@ func qemuAssembler(format string, filename string, uefi bool, imageOptions distr
...
+			"ignition",
```
Comment (mrguitar) [inline internal/distro/rhel8/distro.go:586]: remove ignition please.

File: internal/distro/rhel8/distro.go [hunk 7]
```diff
@@ -531,10 +532,85 @@ func qemuAssembler(format string, filename string, uefi bool, imageOptions distr
...
+			"rsync",
+			"ima-evm-utils",
+			"bash-completion",
+			"tmux", "screen",
```
Comment (mrguitar) [inline internal/distro/rhel8/distro.go:590]: remove screen

File: internal/distro/rhel8/distro.go [hunk 8]
```diff
@@ -531,10 +532,85 @@ func qemuAssembler(format string, filename string, uefi bool, imageOptions distr
...
+			"policycoreutils-python-utils",
+			"setools-console",
+			"audit", "rng-tools",
+			"bluez", "bluez-libs", "bluez-mesh", "wpan-tools",
```
Comment (mrguitar) [inline internal/distro/rhel8/distro.go:594]: bluez-mesh & wpan-tools do not exist - remove or comment.

File: internal/distro/rhel8/distro.go [hunk 9]
```diff
@@ -531,10 +532,85 @@ func qemuAssembler(format string, filename string, uefi bool, imageOptions distr
...
+			"kernel-tools", "libgpiod-utils",
```
Comment (mrguitar) [inline internal/distro/rhel8/distro.go:595]: I don't see libgpiod-utils either

**Comment by teg** [comment]
Thanks for the comments Ben! I'll fix and push a new version.

**Comment by teg** [comment]
This is superseded by #727.

**Pull request closed without merging**